### PR TITLE
d/tfe_project: Output workspace names

### PR DESCRIPTION
**Pull request by 1natedawg**
## Description

This PR is to emit the names of all the workspaces within a project. My use case is is follows:
- I have a known project name and want to act on certain workspaces within it (based on a workspace name suffix)
- I can use tfe_project to look up the IDs of all the workspaces in that project
- I don't have a clean way to look up those workspaces by ID to get their names

So, this allows me to capture the workspace names within a project and act on them. If there is a better alternative, let me know!

_Remember to:_

- [x] _Update the [Change Log](https://github.com/hashicorp/terraform-provider-tfe/blob/main/docs/changelog-process.md)_
- [x] _Update the [Documentation](https://github.com/hashicorp/terraform-provider-tfe/blob/main/docs/changelog-process.md#updating-the-documentation)_

## Testing plan

1. Create a project called `foo`
2. Create a `tfe_project` data source with name `foo`, e.g.:
``` 
data "tfe_project" "foo" {
  organization = "my-org"
  name = "foo"
}

output "project" {
  value = data.tfe_project.foo
}
```
3. Run `terraform plan`/`terraform apply`
4. Output should contain a list called `workspace_names` in addition to the attributes that were already there.

## Output from acceptance tests

_Please run applicable acceptance tests locally and include the output here. See [testing.md](https://github.com/hashicorp/terraform-provider-tfe/blob/main/docs/testing.md) to learn how to run acceptance tests._

_If you are an external contributor, your contribution(s) will first be reviewed before running them against the project's CI pipeline._

```
TESTARGS="-run TestAccTFEProjectDataSource_caseInsensitive" make testacc                                                   [13:00:03]
TF_ACC=1 TF_LOG_SDK_PROTO=OFF go test $(go list ./... |grep -v 'vendor') -v -run TestAccTFEProjectDataSource_caseInsensitive -timeout 15m
?       github.com/hashicorp/terraform-provider-tfe     [no test files]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-tfe/internal/client     0.838s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-tfe/internal/logging    0.749s [no tests to run]
?       github.com/hashicorp/terraform-provider-tfe/internal/provider/validators        [no test files]
?       github.com/hashicorp/terraform-provider-tfe/version     [no test files]
=== RUN   TestAccTFEProjectDataSource_caseInsensitive
--- PASS: TestAccTFEProjectDataSource_caseInsensitive (5.01s)
PASS
ok      github.com/hashicorp/terraform-provider-tfe/internal/provider   5.817s

...
```

### CODE REVIEW

**Review by brandonc — COMMENTED** [review]
This looks like a reasonable change but I want to point out that there is a subtle drawback to having two "Set" type attributes side-by-side like this: if you convert it to a list, the indices may or may not correspond to each other. In other words, `tolist(data.tfe_project.this.workspace_ids)[0]` may or may not correspond to `tolist(data.tfe_project.this.workspace_names)[0]` and so it seems to me like the symmetry suggested by the attributes could introduce some unintended bugs downstream.

**Comment by brandonc** [comment]
It may be acceptable to highlight in the docs that the attributes are sets and cannot be meaningfully mapped to each other.

**Comment by brandonc** [comment]
Also, can you add a test assertion to `TestAccTFEProjectDataSource_basic` ? Thanks for the contribution!

**Comment by 1natedawg** [comment]
Thanks for the feedback @brandonc ! Yeah, I'm with you - I'd love to be able to tie them together, but I couldn't come up with a good way to do so. I've added a test assertion and a note in the docs, let me know if you need any additional changes!

Acceptance tests w/ additional assertion:
```
TESTARGS="-run TestAccTFEProjectDataSource*" make testacc                
TF_ACC=1 TF_LOG_SDK_PROTO=OFF go test $(go list ./... |grep -v 'vendor') -v -run TestAccTFEProjectDataSource* -timeout 15m
?       github.com/hashicorp/terraform-provider-tfe     [no test files]
?       github.com/hashicorp/terraform-provider-tfe/internal/provider/validators        [no test files]
?       github.com/hashicorp/terraform-provider-tfe/version     [no test files]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-tfe/internal/client     0.494s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-tfe/internal/logging    0.287s [no tests to run]
=== RUN   TestAccTFEProjectDataSource_basic
--- PASS: TestAccTFEProjectDataSource_basic (10.95s)
=== RUN   TestAccTFEProjectDataSource_caseInsensitive
--- PASS: TestAccTFEProjectDataSource_caseInsensitive (7.93s)
PASS
ok      github.com/hashicorp/terraform-provider-tfe/internal/provider   19.320s
```

File: website/docs/d/project.html.markdown [hunk 1]
```diff
@@ -9,6 +9,8 @@ Get information on a Project.
 
 Use this data source to get information about a project.
 
+~> **NOTE:** The `workspace_ids` and `workspace_names` attributes are not guaranteed to return values in the same order, so they cannot be reliably mapped to one another. To map workspace names to IDs reliably, it is recommended to pass those names into the `tfe_workspace_ids` data source.
```
Comment (brandonc) [inline website/docs/d/project.html.markdown:12]: 👍

**Review by brandonc — APPROVED** [review]
CHANGELOG Just needs a rebase and I'll be glad to merge it

**Review by brandonc — APPROVED** [review]

**Pull request merged**